### PR TITLE
[app_dart] Add access token authentication for swarming tasks

### PR DESCRIPTION
**Pull request by CaseyHillers**
Switches `SwarmingAuthenticationProvider` id token authentication to access token.

Make some members in `AuthenticationProvider` public to remove code duplication in `SwarmingAuthenticationProvider`.

# Issues
https://github.com/flutter/flutter/issues/66191

# Tests

Switched tests from id token to access token equivalents.

Check that only the LUCI prod account is authenticated.

### CODE REVIEW

File: app_dart/lib/src/request_handling/swarming_authentication.dart [hunk 1]
```diff
@@ -111,9 +100,57 @@ class SwarmingAuthenticationProvider extends AuthenticationProvider {
...
+  /// validated against a different endpoint. We only validate access tokens if they belong
+  /// to a DeviceLab service account.
+  ///
+  /// If LUCI auth adds id tokens, we can switch to that and remove this.
+  Future<AuthenticatedContext> authenticateAccessToken(String accessToken,
+      {ClientContext clientContext, Logging log}) async {
+    // Authenticate as a signed-in Google account via OAuth id token.
+    final HttpClient client = httpClientProvider();
+    try {
+      final HttpClientRequest verifyTokenRequest = await client.getUrl(Uri.https(
```
Comment (CaseyHillers) [inline app_dart/lib/src/request_handling/swarming_authentication.dart:122]: Using the Google OAuth apis was malforming the access token request, causing a 400 bad request to return from the API. I scratched it and went for this approach as it's similar to id tokens.

File: app_dart/lib/src/request_handling/swarming_authentication.dart [hunk 2]
```diff
@@ -111,9 +100,57 @@ class SwarmingAuthenticationProvider extends AuthenticationProvider {
...
 
     throw const Unauthenticated('Request rejected due to not from LUCI or Cocoon agent');
   }
+
+  /// Authenticate [accessToken] against Google OAuth 2 API.
+  ///
+  /// Access tokens are the legacy authentication strategy for Google OAuth, where ID tokens
+  /// are the new technique to use. LUCI auth only generates access tokens, and must be
+  /// validated against a different endpoint. We only validate access tokens if they belong
+  /// to a DeviceLab service account.
```
Comment (godofredoc) [inline app_dart/lib/src/request_handling/swarming_authentication.dart:114]: Should this be updated to also cover the prod luci service account?

File: app_dart/lib/src/request_handling/swarming_authentication.dart (hunk 1 above)
Comment (godofredoc) [reply app_dart/lib/src/request_handling/swarming_authentication.dart:122]: Is this an error with the APIs or the client libraries making the request?

File: app_dart/lib/src/request_handling/swarming_authentication.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handling/swarming_authentication.dart:114]: Done

File: app_dart/lib/src/request_handling/swarming_authentication.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handling/swarming_authentication.dart:122]: The client libraries as the the request looks to be putting all possible query parameters in. I'm not sure who owns that library, but to unblock for now I went with this approach.

Comment (godofredoc) [reply app_dart/lib/src/request_handling/swarming_authentication.dart:122]: Actually the question was to decide if we need to open a bug for the client library but not a blocker for this PR.

**Review by godofredoc — APPROVED** [review]

File: app_dart/lib/src/request_handling/swarming_authentication.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handling/swarming_authentication.dart:122]: I'll create a reproduction and file an issue.

**Pull request merged**